test/checkpoint.test.js

```diff
@@ -1,28 +1,43 @@
 var async = require('async');
 var loopback = require('../');
-
-// create a unique Checkpoint model
-var Checkpoint = loopback.Checkpoint.extend('TestCheckpoint');
-
-var memory = loopback.createDataSource({
-  connector: loopback.Memory
-});
-Checkpoint.attachTo(memory);
+var expect = require('chai').expect;
 
 describe('Checkpoint', function() {
-  describe('current()', function() {
+  var Checkpoint;
+  describe('bump and current', function() {
+    beforeEach(function() {
+      // create a unique Checkpoint model
+      Checkpoint = loopback.Checkpoint.extend('TestCheckpoint');
+      var memory = loopback.createDataSource({
+        connector: loopback.Memory
+      });
+      Checkpoint.attachTo(memory);
+    });
     it('returns the highest `seq` value', function(done) {
       async.series([
-        Checkpoint.create.bind(Checkpoint),
-        Checkpoint.create.bind(Checkpoint),
+        Checkpoint.bump.bind(Checkpoint),
+        Checkpoint.bump.bind(Checkpoint),
         function(next) {
           Checkpoint.current(function(err, seq) {
-            if (err) next(err);
             expect(seq).to.equal(3);
-            next();
+            next(err);
           });
         }
       ], done);
     });
+
+    it('`Checkpoint.current()` for non existing `checkpoint` should initialize `checkpoint`', function(done) {
+      Checkpoint.current(function(err, seq) {
+        expect(seq).to.equal(1);
+        done(err);
+      });
+    });
+
+    it('`bump()` for non existing `checkpoint` should initialize `checkpoint`', function(done) {
+      Checkpoint.bump(function(err, cp) {
+        expect(cp.seq).to.equal(2);
```

Comment:
Hmm, I find this confusing from reader's point of view. Could you please add a comment explaining why `seq` should be `2`? Also perhaps modify test name to something like `bump() works when singleton instance does not exists yet`